pywps/Process/InAndOutputs.py

```diff
@@ -1014,12 +1014,10 @@ def setValue(self, value):
         elif value.__class__.__name__=='StringIO' or value.__class__.__name__=='StringO':
             import tempfile
             from os import curdir
-            stringIOName = tempfile.mkstemp(prefix="pywpsOutput",dir=curdir) #(5, '/tmp/pywps-instanceS2j6ve/pywpsOutputZxSM6V')
-            stringIOName=stringIOName[1]
-
-            stringIOFile=open(stringIOName,"w")
-            stringIOFile.write(value.getvalue())
-            stringIOFile.close()
+            fh, stringIOName = tempfile.mkstemp(prefix="pywpsOutput",
+                                                dir=curdir) #(5, '/tmp/pywps-instanceS2j6ve/pywpsOutputZxSM6V')
+            fh.write(value.getvalue())
+            fh.close()
             self.value=stringIOName
```

Comment:
I concur that the dangling file handles are a legitimate problem, but does this solution actually work?  `fh`, the file handle, is simply an integer.  In my standalone testing with Python 2.6 and 2.7 `fh.write()` and `fh.close()` fail.  Seems like you either need to use: 1)  `fdopen()` to open a file from the descriptor and then write and close the file, or 2) `os.write()` and `os.close()` to operate directly on the file descriptor.

References:
http://stackoverflow.com/questions/7799680/python-need-to-close-file-from-mkstemp-if-i-use-fdopen
http://stackoverflow.com/questions/168559/python-how-do-i-convert-an-os-level-handle-to-an-open-file-to-a-file-object